### PR TITLE
Update powsybl-core dependency

### PR DESCRIPTION
**Pull request by rolnico**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**
Dependencies upgrade

**What is the current behavior?**
HPC uses powsybl-core 4.1

**What is the new behavior (if this is a feature change)?**
powsybl-parent is bumped to v18
HPC uses powsybl-core 6.3.0
Java version is specified and switched to Java 17
junit version is specified (5.10.1)
Mockito is bumped to 5.10.0.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmStatusManager.java [hunk 1]
```diff
@@ -25,7 +25,7 @@ class SlurmStatusManager {
     private static final String TIME_CMD = "date +\"%Y-%m-%d %H:%M:%S\"";
     private static final String INFO_CORES_CMD = "sinfo -h -o %C";
     private static final String QUEUE_CORES_PER_JOB = "squeue -h --format=\"%C %j\"";
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss\n");
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss\n").withZone(ZoneOffset.UTC);
```
Comment (olperr1) [inline computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmStatusManager.java:28]: Should we really use the UTC zone here? Isn't it better to use the system zone?

If this latter option is adopted, the following elements should be changed as:
```java
TIME_CMD = "date +\"%Y-%m-%d %H:%M:%S %Z\"";
DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss zzz\n")
```

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmStatusManager.java (hunk 1 above)
Comment (rolnico) [reply computation-slurm/src/main/java/com/powsybl/computation/slurm/SlurmStatusManager.java:28]: I personally find it better to always use the UTC timezone in order to be sure to be consistent everywhere.
Your solution however also works.

File: computation-mpi/src/main/java/com/powsybl/computation/mpi/MpiJobSchedulerImpl.java [hunk 2]
```diff
@@ -488,13 +488,14 @@ private void processCompletedTasks(MpiJob job, List<MpiTask> completedTasks) thr
...
-                long taskDurationSeenByMaster = new Duration(task.getStartTime(), task.getEndTime()).getMillis();
+                // duration of the task seen by the master in ns
+                long taskDurationSeenByMaster = Duration.between(task.getStartTime(), task.getEndTime()).getNano();
 
                 // decode task result messages
                 Messages.TaskResult message = Messages.TaskResult.parseFrom(task.getResultMessage());
 
                 // duration of the task seen by the slave in ms
+                // TODO : duration is now in ns and not in ms
                 long taskDurationSeenBySlave = message.getTaskDuration(); // number of ms
```
[the remark this answers is not in the document]
Comment (rolnico) [reply computation-mpi/src/main/java/com/powsybl/computation/mpi/MpiJobSchedulerImpl.java:499]: The thing is, I don't really know how things work in Messages.java, I'm not sure the duration is in ns (it was not before the PR)

File: computation-slurm/src/test/java/com/powsybl/computation/slurm/ScontrolCmdTest.java [hunk 3]
```diff
@@ -58,68 +59,70 @@ public void test() throws SlurmCmdNonZeroException {
...
+               TRES=cpu=8,node=1,billing=8
+               Socks/Node=* NtasksPerN:B:S:C=0:0:*:* CoreSpec=*
+               MinCPUsNode=1 MinMemoryNode=0 MinTmpDiskNode=0
+               Features=(null) DelayBoot=00:00:00
+               OverSubscribe=YES Contiguous=0 Licenses=(null) Network=(null)
+               Command=/home/test/yichen_slurm_unit/unit_test_3609105074134694169/cmdId.batch
+               WorkDir=/home/test/yichen_slurm_unit/unit_test_3609105074134694169
+               StdErr=/home/test/yichen_slurm_unit/unit_test_3609105074134694169/cmdId_1.err
+               StdIn=/dev/null
+               StdOut=/home/test/yichen_slurm_unit/unit_test_3609105074134694169/cmdId_1.out
```
[the remark this answers is not in the document]
Comment (rolnico) [reply computation-slurm/src/test/java/com/powsybl/computation/slurm/ScontrolCmdTest.java]: done

File: computation-mpi/src/main/java/com/powsybl/computation/mpi/MpiJobSchedulerImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolnico) [reply computation-mpi/src/main/java/com/powsybl/computation/mpi/MpiJobSchedulerImpl.java]: durations are back in ms

File: computation-mpi/src/main/proto/messages.proto [hunk 4]
```diff
@@ -59,7 +61,7 @@ message Task {
     message Command {
         required string program = 1;
         repeated string argument = 2;
-	optional int32 timeout = 3;
+	      optional int32 timeout = 3;
```
Comment (olperr1) [inline computation-mpi/src/main/proto/messages.proto:64]: Could you fix the indent on this line?

**Review by olperr1 — CHANGES_REQUESTED** [review]

**Comment by rolnico** [comment]
Warning : slurm integration tests must be done locally, using a server with SLURM installed

**Review by olperr1 — CHANGES_REQUESTED** [review]
Could you add an explanation for the disabled tests?
```patch
-@Disabled
+@Disabled("Slurm integration tests must be run locally.")
```

**Review by olperr1 — CHANGES_REQUESTED** [review]
I think it would be best to revert using powsybl-core 6.2.2 because of the CGMES import performance issue.

**Comment by rolnico** [comment]
> I think it would be best to revert using powsybl-core 6.2.2 because of the CGMES import performance issue.

Done

**Review by olperr1 — APPROVED** [review]
There's still a problem with the test coverage, but appart that it seems ok to me.

**Pull request merged**